File: MAStaff-Bungee/src/main/java/es/angelillo15/mast/bungee/listener/redis/staff/OnStaffTalk.java
package es.angelillo15.mast.bungee.listener.redis.staff;

import es.angelillo15.mast.api.redis.EventHandler;
import es.angelillo15.mast.api.redis.Listener;
import es.angelillo15.mast.api.redis.events.staff.StaffChatMessageEvent;
import es.angelillo15.mast.bungee.MAStaff;
import es.angelillo15.mast.bungee.config.Messages;
import es.angelillo15.mast.bungee.utils.StaffUtils;

public class OnStaffTalk implements Listener {
    @EventHandler
    public void onStaffTalk(StaffChatMessageEvent event) {
        String message = Messages.GET_STAFF_CHAT_FORMAT()
                .replace("{server}", event.getServer())
                .replace("{player}", event.getStaffName())
                .replace("{message}", event.getStaffMessage());

        StaffUtils.sendStaffChatMessage(message);
    }
}

File: MAStaff-Bungee/src/main/java/es/angelillo15/mast/bungee/listener/staffchat/OnStaffTalk.kt
package es.angelillo15.mast.bungee.listener.staffchat

import es.angelillo15.mast.handler.OnStaffMessageEvent
import net.md_5.bungee.api.connection.ProxiedPlayer
import net.md_5.bungee.api.event.ChatEvent
import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler

class OnStaffTalk : Listener, OnStaffMessageEvent() {
    @EventHandler
    fun onStaffTalk(event: ChatEvent) {
        if (event.isCommand) return
        if (event.sender !is ProxiedPlayer) return
        val player = event.sender as ProxiedPlayer

        event.isCancelled = onStaffMessageEvent(player.name, event.message, player.server.info.name)
    }
}